musicbot/bot.py

```diff
@@ -388,7 +388,8 @@ def _delete_old_audiocache(self, path=AUDIO_CACHE_PATH):
 
                 async for lmsg in self.logs_from(channel, limit=1):
                     if lmsg != last_np_msg and last_np_msg:
-                        await self.safe_delete_message(last_np_msg)
+                        if self.config.delete_nowplaying:
```

Comment:
I've no idea what's going on in the rest of the code so I didn't want to if-it-out in case it's important...

387: `if last_np_msg and last_np_msg.channel == channel and self.config.delete_nowplaying:`
I can quickly commit that if that's better.